h/static/scripts/controllers/search-bar-controller.js

```diff
@@ -0,0 +1,151 @@
+'use strict';
+
+var Controller = require('../base/controller');
+var setElementState = require('../util/dom').setElementState;
+
+/**
+ * Controller for the search bar.
+ */
+class SearchBarController extends Controller {
+  constructor(element) {
+    super(element);
+
+    this._input = this.refs.searchBarInput;
+    this._dropdown = this.refs.searchBarDropdown;
+    this._dropdownItems = Array.from(element.querySelectorAll('[data-ref="searchBarDropdownItem"]'));
+
+    var clearActiveDropdownItem = () => {
+      var activeItem = element.querySelector('.js-search-bar-dropdown-menu-item--active');
+      if (activeItem) {
+        activeItem.classList.remove('js-search-bar-dropdown-menu-item--active');
+      }
+    };
+
+    var updateActiveDropdownItem = element => {
+      clearActiveDropdownItem();
+      element.classList.add('js-search-bar-dropdown-menu-item--active');
+    };
+
+    var selectFacet = facet => {
+      this._input.value = this._input.value + facet;
+
+      closeDropdown();
+
+      setTimeout(function() {
+        this._input.focus();
+      }.bind(this), 0);
+    };
+
+    var getPreviousSiblingElement = element => {
+        if (!element) {
+          return null;
+        }
+        do {
+          element = element.previousSibling;
+        } while (element && element.nodeType != 1);
+        return element;
+    };
+
+    var getNextSiblingElement = element => {
+        if (!element) {
+          return null;
+        }
+        do {
+          element = element.nextSibling;
+        } while (element && element.nodeType != 1);
+        return element;
+    };
+
+    var setupListenerKeys = event => {
+      var activeItem = element.querySelector('.js-search-bar-dropdown-menu-item--active');
+
+      switch (event.keyCode) {
+      case 13:
+        // enter key
+        event.preventDefault();
+
+        if (activeItem) {
+          var facet = activeItem.querySelector('[data-ref="searchBarDropdownItemTitle"]').innerHTML.trim();
+          selectFacet(facet);
+        }
+        break;
+      case 38:
+        // up arrow key
+        updateActiveDropdownItem(getPreviousSiblingElement(activeItem) || this._dropdownItems[this._dropdownItems.length - 1]);
+        break;
+      case 40:
+        // down arrow key
+        updateActiveDropdownItem(getNextSiblingElement(activeItem) || this._dropdownItems[0]);
+        break;
+      default:
+        break;
```

Comment:
I think instead of a long `switch` (which will potentially grow longer in the future) this might be better as a separate function for handling each key:

```javascript
var handlers = {};

var handleEnterKey = event => { ...};
handlers.ENTER_KEY_CODE = handleEnterKey;

var handleUpArrow = event => { ...};
handlers.UP_ARROW_KEY_CODE = handleUpArrow;

...

handlers[event.keyCode](event);
```

This would also mean that the handler functions could easily be called directly, for easy testing.